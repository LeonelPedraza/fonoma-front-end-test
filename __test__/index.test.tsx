import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Home from '@/pages/index'

describe('Render home page', () => {
  it('Should have a heading', () => {
    render(<Home coins={null} />)

    const heading = screen.getByRole('heading', {
      name: 'Exchange rate calculator',
    })

    expect(heading).toBeInTheDocument()
  })

  it('Should have a submit button', () => {
    render(<Home coins={null} />)

    const button = screen.getByRole('button', {
      name: 'Convert',
    })

    expect(button).toBeInTheDocument()
  })
})

describe('Render homepage', () => {
  it('Should render homepage', () => {
    const { container } = render(<Home coins={null}/>)
    expect(container).toMatchSnapshot()
  })
})