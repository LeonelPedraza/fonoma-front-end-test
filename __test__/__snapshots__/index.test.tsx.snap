// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Should render homepage unchanged 1`] = `
<div>
  <main
    class="sc-bgqQcB gGNHbj"
  >
    <form
      class="sc-gTRrQi gLAJmf"
    >
      <h1
        class="sc-gxYJeL hRdfyV"
      >
        Exchange rate calculator
      </h1>
      <div
        class="sc-ispOId cAbXnt"
      >
        <div
          class="sc-kcuKUB kIUYkV"
        >
          <label
            class="sc-fFGjHI kDjevi"
          >
            From:
          </label>
          <select
            class="sc-dcrmVg kgKaAH"
            id="select-coin-from"
            name="from"
          />
        </div>
        <div
          class="sc-kcuKUB kIUYkV"
        >
          <label
            class="sc-fFGjHI kDjevi"
          >
            To:
          </label>
          <select
            class="sc-dcrmVg kgKaAH"
            data-testid="selectCoin"
            name="to"
          />
        </div>
      </div>
      <div
        class="sc-kuWgmH cpPOIM"
      >
        <label
          class="sc-fFGjHI kDjevi"
        >
          Amount:
        </label>
        <input
          class="sc-duSInm gFBLF"
          min="1"
          name="amount"
          placeholder="Amount"
          step="any"
          type="number"
        />
        <label
          class="sc-fFGjHI sc-dicizt kDjevi hqhyWF"
        />
      </div>
      <div
        class="sc-ewnqHT jPedlw"
      >
        <button
          class="sc-iVCKna jEIhLQ"
          type="submit"
        >
          Convert
        </button>
      </div>
    </form>
  </main>
</div>
`;

exports[` renders homepage unchanged 1`] = `
<div>
  <main
    class="sc-bgqQcB gGNHbj"
  >
    <form
      class="sc-gTRrQi gLAJmf"
    >
      <h1
        class="sc-gxYJeL hRdfyV"
      >
        Exchange rate calculator
      </h1>
      <div
        class="sc-ispOId cAbXnt"
      >
        <div
          class="sc-kcuKUB kIUYkV"
        >
          <label
            class="sc-fFGjHI kDjevi"
          >
            From:
          </label>
          <select
            class="sc-dcrmVg kgKaAH"
            id="select-coin-from"
            name="from"
          />
        </div>
        <div
          class="sc-kcuKUB kIUYkV"
        >
          <label
            class="sc-fFGjHI kDjevi"
          >
            To:
          </label>
          <select
            class="sc-dcrmVg kgKaAH"
            data-testid="selectCoin"
            name="to"
          />
        </div>
      </div>
      <div
        class="sc-kuWgmH cpPOIM"
      >
        <label
          class="sc-fFGjHI kDjevi"
        >
          Amount:
        </label>
        <input
          class="sc-duSInm gFBLF"
          min="1"
          name="amount"
          placeholder="Amount"
          step="any"
          type="number"
        />
        <label
          class="sc-fFGjHI sc-dicizt kDjevi hqhyWF"
        />
      </div>
      <div
        class="sc-ewnqHT jPedlw"
      >
        <button
          class="sc-iVCKna jEIhLQ"
          type="submit"
        >
          Convert
        </button>
      </div>
    </form>
  </main>
</div>
`;

exports[`Render home page renders homepage unchanged 1`] = `
<div>
  <main
    class="sc-bgqQcB gGNHbj"
  >
    <form
      class="sc-gTRrQi gLAJmf"
    >
      <h1
        class="sc-gxYJeL hRdfyV"
      >
        Exchange rate calculator
      </h1>
      <div
        class="sc-ispOId cAbXnt"
      >
        <div
          class="sc-kcuKUB kIUYkV"
        >
          <label
            class="sc-fFGjHI kDjevi"
          >
            From:
          </label>
          <select
            class="sc-dcrmVg kgKaAH"
            id="select-coin-from"
            name="from"
          />
        </div>
        <div
          class="sc-kcuKUB kIUYkV"
        >
          <label
            class="sc-fFGjHI kDjevi"
          >
            To:
          </label>
          <select
            class="sc-dcrmVg kgKaAH"
            data-testid="selectCoin"
            name="to"
          />
        </div>
      </div>
      <div
        class="sc-kuWgmH cpPOIM"
      >
        <label
          class="sc-fFGjHI kDjevi"
        >
          Amount:
        </label>
        <input
          class="sc-duSInm gFBLF"
          min="1"
          name="amount"
          placeholder="Amount"
          step="any"
          type="number"
        />
        <label
          class="sc-fFGjHI sc-dicizt kDjevi hqhyWF"
        />
      </div>
      <div
        class="sc-ewnqHT jPedlw"
      >
        <button
          class="sc-iVCKna jEIhLQ"
          type="submit"
        >
          Convert
        </button>
      </div>
    </form>
  </main>
</div>
`;

exports[`Render homepage Should render homepage 1`] = `
<div>
  <main
    class="sc-bgqQcB gGNHbj"
  >
    <form
      class="sc-gTRrQi gLAJmf"
    >
      <h1
        class="sc-gxYJeL hRdfyV"
      >
        Exchange rate calculator
      </h1>
      <div
        class="sc-ispOId cAbXnt"
      >
        <div
          class="sc-kcuKUB kIUYkV"
        >
          <label
            class="sc-fFGjHI kDjevi"
          >
            From:
          </label>
          <select
            class="sc-dcrmVg kgKaAH"
            id="select-coin-from"
            name="from"
          />
        </div>
        <div
          class="sc-kcuKUB kIUYkV"
        >
          <label
            class="sc-fFGjHI kDjevi"
          >
            To:
          </label>
          <select
            class="sc-dcrmVg kgKaAH"
            data-testid="selectCoin"
            name="to"
          />
        </div>
      </div>
      <div
        class="sc-kuWgmH cpPOIM"
      >
        <label
          class="sc-fFGjHI kDjevi"
        >
          Amount:
        </label>
        <input
          class="sc-duSInm gFBLF"
          min="1"
          name="amount"
          placeholder="Amount"
          step="any"
          type="number"
        />
        <label
          class="sc-fFGjHI sc-dicizt kDjevi hqhyWF"
        />
      </div>
      <div
        class="sc-ewnqHT jPedlw"
      >
        <button
          class="sc-iVCKna jEIhLQ"
          type="submit"
        >
          Convert
        </button>
      </div>
    </form>
  </main>
</div>
`;

exports[`Render homepage unchanged Should render homepage unchanged 1`] = `
<div>
  <main
    class="sc-bgqQcB gGNHbj"
  >
    <form
      class="sc-gTRrQi gLAJmf"
    >
      <h1
        class="sc-gxYJeL hRdfyV"
      >
        Exchange rate calculator
      </h1>
      <div
        class="sc-ispOId cAbXnt"
      >
        <div
          class="sc-kcuKUB kIUYkV"
        >
          <label
            class="sc-fFGjHI kDjevi"
          >
            From:
          </label>
          <select
            class="sc-dcrmVg kgKaAH"
            id="select-coin-from"
            name="from"
          />
        </div>
        <div
          class="sc-kcuKUB kIUYkV"
        >
          <label
            class="sc-fFGjHI kDjevi"
          >
            To:
          </label>
          <select
            class="sc-dcrmVg kgKaAH"
            data-testid="selectCoin"
            name="to"
          />
        </div>
      </div>
      <div
        class="sc-kuWgmH cpPOIM"
      >
        <label
          class="sc-fFGjHI kDjevi"
        >
          Amount:
        </label>
        <input
          class="sc-duSInm gFBLF"
          min="1"
          name="amount"
          placeholder="Amount"
          step="any"
          type="number"
        />
        <label
          class="sc-fFGjHI sc-dicizt kDjevi hqhyWF"
        />
      </div>
      <div
        class="sc-ewnqHT jPedlw"
      >
        <button
          class="sc-iVCKna jEIhLQ"
          type="submit"
        >
          Convert
        </button>
      </div>
    </form>
  </main>
</div>
`;
